plugins/liquid/refined-types-plugin/src/main/kotlin/arrow/meta/plugins/liquid/phases/analysis/solver/ConstraintChecks.kt

```diff
@@ -491,29 +496,42 @@ data class ControlFlowFn(
 }
 
 /**
- * Special treatment for special control flow functions ([also], [apply], [let], [run])
+ * Special treatment for special control flow functions ([also], [apply], [let], [run], [with])
+ * https://kotlinlang.org/docs/scope-functions.html#function-selection
  */
 private fun controlFlowAnyFunction(
   resolvedCall: ResolvedCall<out CallableDescriptor>
 ): ControlFlowFn? {
-  val thisElement = resolvedCall.arg("this")
+  val thisElement = resolvedCall.arg("this") ?: resolvedCall.arg("receiver")
   val blockElement = resolvedCall.arg("block") as? KtLambdaExpression
   val bodyElement = blockElement?.bodyExpression
-  return if (thisElement != null && blockElement != null && bodyElement != null) {
-    when (resolvedCall.resultingDescriptor.fqNameSafe) {
-      FqName("kotlin.also") -> {
-        val argumentName = blockElement.valueParameters.getOrNull(0)?.name ?: "it"
-        ControlFlowFn(thisElement, bodyElement, argumentName, ControlFlowFn.ReturnBehavior.RETURNS_ARGUMENT)
+  return if (blockElement != null && bodyElement != null) {
+    if (thisElement != null) {
+      when (resolvedCall.resultingDescriptor.fqNameSafe) {
+        FqName("kotlin.also") -> {
+          val argumentName = blockElement.valueParameters.getOrNull(0)?.name ?: "it"
+          ControlFlowFn(thisElement, bodyElement, argumentName, ControlFlowFn.ReturnBehavior.RETURNS_ARGUMENT)
+        }
+        FqName("kotlin.apply") ->
```

Comment:
Do we have some kind of a plan to support lambda args and be able to define these special cases also as Laws?

Reply:
Even if we did so, I think the language won't be expressive enough to implement something like `also`, where you want the function to be executed, yet the result is equal to the receiver. The closest I could think of is analyzing the inlined body, which could work quite well in these cases.